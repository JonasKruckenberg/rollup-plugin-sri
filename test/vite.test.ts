import { build } from 'vite'
import { join } from 'path'
import sri from '../index'
import { FILE } from 'node:dns';
import { fileURLToPath } from 'node:url';
import { OutputAsset } from 'rollup';

process.chdir(join(__dirname, 'fixtures'));

describe('vite', () => {
    test('basic', async () => {
        const output = await build({
            logLevel: 'warn',
            plugins: [{
                enforce: 'post',
                ...sri({ publicPath: '/' })
            }]
        })

        const source = (Array.isArray(output) ? output[0].output : output.output)
            .filter((file): file is OutputAsset => file.type === 'asset')
            .map(({ fileName, source }) => ({ fileName, source }))
            .find(file => file.fileName.endsWith('.html'))

        expect(source).toMatchSnapshot()
    })
})