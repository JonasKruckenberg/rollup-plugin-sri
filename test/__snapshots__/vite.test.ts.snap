// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vite basic 1`] = `
Object {
  "fileName": "index.html",
  "source": "<!DOCTYPE html><html lang=\\"en\\"><head>
    <meta charset=\\"UTF-8\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <title>Document</title>
<script type=\\"module\\" crossorigin=\\"anonymous\\" src=\\"/assets/index.aa78867d.js\\" integrity=\\"sha384-mB+SG9rRCWqiveE+JXQxMJkGnWb3UovH9/Y5ohGCtOlS5E2Z3R3RTq28TldJL7/0\\"></script>
</head>
<body>
    

</body></html>",
}
`;
